src/core/components/button/Button.stories.tsx

```diff
@@ -0,0 +1,262 @@
+import { Button, ButtonProps } from './Button';
+import type { ButtonPriority } from './types';
+import {
+	buttonBrand,
+	buttonBrandAlt,
+	buttonReaderRevenue,
+	buttonReaderRevenueBrand,
+	buttonReaderRevenueBrandAlt,
+} from './index';
+import { SvgCross } from '@guardian/src-icons';
+import { asPlayground, asChromaticStory } from '../../../lib/story-intents';
+import type {
+	Story,
+	Parameters,
+} from '../../../@types/storybook-emotion-10-fixes';
+// These types are the right types, but don't work with Storybook v6 which uses Emotion v10
+// import type { Args, Story } from '@storybook/react';
+
+const priorityArgs: ButtonPriority[] = [
+	'primary',
+	'secondary',
+	'tertiary',
+	'subdued',
+];
+const readerRevenuePriorityArgs: ButtonPriority[] = ['primary', 'tertiary'];
+const themeParameters = {
+	default: {},
+	brand: {
+		backgrounds: {
+			default: 'brandBackground.primary',
+		},
+		theme: buttonBrand,
+	},
+	brandAlt: {
+		backgrounds: {
+			default: 'brandAltBackground.primary',
+		},
+		theme: buttonBrandAlt,
+	},
+	readerRevenue: {
+		theme: buttonReaderRevenue,
+	},
+	readerRevenueBrand: {
+		backgrounds: {
+			default: 'brandBackground.primary',
+		},
+		theme: buttonReaderRevenueBrand,
+	},
+	readerRevenueBrandAlt: {
+		backgrounds: {
+			default: 'brandAltBackground.primary',
+		},
+		theme: buttonReaderRevenueBrandAlt,
+	},
+};
+const createStory = (args: Partial<ButtonProps>, parameters: Parameters) => {
+	const story = Template.bind({});
+
+	story.parameters = parameters;
+	story.args = args;
+	asChromaticStory(story);
+
+	return story;
+};
```

Comment:
This bit of repetition was really annoying for the themed priority stories, although it could technically be used by all stories in this module. Not sure if it goes too far, or not far enough!